providers/azure/azure.go

```diff
@@ -42,16 +42,17 @@ var DefaultConfig = Config{
 
 // Config Azure storage configuration.
 type Config struct {
-	StorageAccountName string             `yaml:"storage_account"`
-	StorageAccountKey  string             `yaml:"storage_account_key"`
-	ContainerName      string             `yaml:"container"`
-	Endpoint           string             `yaml:"endpoint"`
-	MaxRetries         int                `yaml:"max_retries"`
-	MSIResource        string             `yaml:"msi_resource"`
-	UserAssignedID     string             `yaml:"user_assigned_id"`
-	PipelineConfig     PipelineConfig     `yaml:"pipeline_config"`
-	ReaderConfig       ReaderConfig       `yaml:"reader_config"`
-	HTTPConfig         exthttp.HTTPConfig `yaml:"http_config"`
+	StorageAccountName string `yaml:"storage_account"`
+	StorageAccountKey  string `yaml:"storage_account_key"`
+	ContainerName      string `yaml:"container"`
+	Endpoint           string `yaml:"endpoint"`
+	MaxRetries         int    `yaml:"max_retries"`
+	// Deprecated: Is automatically set by the Azure SDK.
+	MSIResource    string             `yaml:"msi_resource"`
```

Comment:
I think we planned to log warning when someone provides it? Let's do that in next PR if needed (: